### PR TITLE
[PREVIEW] - Update health check for IDAM

### PR DESCRIPTION
**Pull request by dtaylor7**
# Description

[DIV-3774 - Update health check for IDAM on DN](https://tools.hmcts.net/jira/browse/DIV-3774)

- Add proxy to enable health check on AKS
- Refactor healthcheck tests
- Add missing test for `case-orchistration-service` health check test

### CODE REVIEW

File: middleware/healthcheck.js [hunk 1]
```diff
@@ -14,22 +15,33 @@ const options = {
...
           logger.error(`Health check failed on redis: ${error}`);
-          return false;
+          return healthcheck.status(false);
         });
     }),
-    'idam-authentication': healthcheck.web(config.services.idam.authenticationHealth, {
-      callback: (error, res) => { // eslint-disable-line id-blacklist
-        if (error) {
+    'idam-authentication': healthcheck.raw(() => {
+      const proxyRequest = request
```
Comment (alexwakeman) [inline middleware/healthcheck.js]: Nit-picky comment - why do you set a default when you only use the request object once? May as well just pass the proxy in the options, and do a straightforward `request(options)`

**Comment by alexwakeman** [comment]
As mentioned in RFE PR for this - the proxy settings won't work in the non-AKS environment so will need to switch between them cc @dtaylor7 @mrganeshraja

File: middleware/healthcheck.js (hunk 1 above)
Comment (dtaylor7) [reply middleware/healthcheck.js]: yup good point

**Comment by dtaylor7** [comment]
@alexwakeman resolved, thanks for heads up

**Pull request merged**